File: json_to_flutter/models/text.dart
import 'index.dart';
import 'submodels/shadow.dart';
import '../functions/util.dart';

class Text extends Widget {
  double rotation;
  String type;
  String text;
  bool withAreaBox;
  bool withColor;
  String color;
  String textAlign;
  bool underline;
  bool strikethrough;
  String fontFamily;
  String fontWeight;
  double fontSize;
  double opacity;
  Shadow shadow;
  Text(double x, double y, double w, double h, double gw, double gh,
      {String id = "-1"})
      : super(x, y, w, h, gw, gh, id);

  @override
  String generateWidget(No no) {
    return """
    Text(
      "$text",
      overflow: TextOverflow.ellipsis,
      style: TextStyle(
        fontFamily: "$fontFamily",
        fontSize: ${sz(fontSize)},
        ${_fontWeight()}
        $color
        ${_decoration()}
      ),
    )
    """;
    //${shadow.print(isText: true)}
  }

  Text.fromJson(Map<String, dynamic> json)
      : super(
            json["x"].toDouble(),
            json["y"].toDouble(),
            json["w"].toDouble(),
            json["h"].toDouble(),
            json["globalW"].toDouble(),
            json["globalH"].toDouble(),
            json["name"]) {
    type = json['type'];
    text = json['text'].replaceAll("\n", "\\n");
    id = text;
    withAreaBox = json['withAreaBox'];
    withColor = json['wcolor'];
    rotation = fixDouble(json['rotation']);
    opacity = fixDouble(json['opacity'].toDouble());
    color = hexColorToFlutterColor(
        json['color'].toString(), withColor ? opacity : 0);
    textAlign = json['textAlign'];
    underline = json['underline'];
    strikethrough = json['strikethrough'];
    fontFamily = json['fontFamily'];
    fontWeight = json['fontWeight'];
    fontSize = fixDouble(json['fontSize'].toDouble());
    shadow = json['shadow'] != null
        ? new Shadow.fromJson(json['shadow'], withColor)
        : null;
  }
  String _decoration() {
    if (strikethrough) {
      return "decoration: TextDecoration.underline,";
    } else if (underline) {
      return "decoration: TextDecoration.underline,";
    }
    return "";
  }

  String _fontWeight() {
    fontWeight = fontWeight.toLowerCase().replaceAll("-", "");
    if (fontWeight == "thin") {
      fontWeight = "100";
    } else if (fontWeight == "extraligth") {
      fontWeight = "200";
    } else if (fontWeight == "light") {
      fontWeight = "300";
    } else if (fontWeight == "medium") {
      fontWeight = "500";
    } else if (fontWeight == "semibold") {
      fontWeight = "600";
    } else if (fontWeight == "bold") {
      fontWeight = "700";
      return 'fontWeight: FontWeight.bold,';
    } else if (fontWeight == "extrabold") {
      fontWeight = "800";
    } else if (fontWeight == "black") {
      fontWeight = "900";
    } else {
      fontWeight = "400";
      return "";
    }
    return 'fontWeight: FontWeight.w${fontWeight},';
  }
}

File: xd_to_flutter_final/src/functions/widget/models/text.js
const { fixDouble, hexColorToFlutterColor, sz } = require("../util");
const { Shadow } = require("./submodels/shadow");

class Text {
  constructor(json) {
    this.type = Text;
    this.x = parseFloat(json["x"]);
    this.y = parseFloat(json["y"]);
    this.w = parseFloat(json["w"]);
    this.h = parseFloat(json["h"]);
    this.gw = parseFloat(json["globalW"]);
    this.gh = parseFloat(json["globalH"]);
    this.type = json['type'];
    this.text = json['text'].replace("\n", "\\n");
    this.id = this.text;
    this.withAreaBox = json['withAreaBox'];
    this.withColor = json['wcolor'];
    this.rotation = fixDouble(json['rotation']);
    this.opacity = fixDouble(json['opacity']);
    this.color = hexColorToFlutterColor(
      json['color'].toString(), this.withColor ? this.opacity : 0);
    this.textAlign = json['textAlign'];
    this.underline = json['underline'];
    this.strikethrough = json['strikethrough'];
    this.fontFamily = json['fontFamily'];
    this.fontWeight = json['fontWeight'];
    this.fontSize = fixDouble(json['fontSize']);
    this.shadow = json['shadow'] != null
      ? new Shadow(json['shadow'], this.withColor)
      : null;
  }

  generateWidget(no) {
    return `
    Text(
      "${this.text}",
      overflow: TextOverflow.ellipsis,
      style: TextStyle(
        fontFamily: "${this.fontFamily}",
        fontSize: ${sz(this.fontSize)},
        ${this._fontWeight()}
        ${this.color}
        ${this._decoration()}
      ),
    )
    `;
    //${shadow.print(isText: true)}
  }

  _decoration() {
    if (this.strikethrough && this.underline) {
      console.log("strikethrough && underline");
    } else if (this.strikethrough) {
      return "decoration: TextDecoration.underline,";
    } else if (this.underline) {
      return "decoration: TextDecoration.underline,";
    }
    return "";
  }

  _fontWeight() {
    this.fontWeight = this.fontWeight.toLowerCase().replace("-", "");
    if (this.fontWeight == "thin") {
      this.fontWeight = "100";
    } else if (this.fontWeight == "extraligth") {
      this.fontWeight = "200";
    } else if (this.fontWeight == "light") {
      this.fontWeight = "300";
    } else if (this.fontWeight == "medium") {
      this.fontWeight = "500";
    } else if (this.fontWeight == "semibold") {
      this.fontWeight = "600";
    } else if (this.fontWeight == "bold") {
      this.fontWeight = "700";
      return 'fontWeight: FontWeight.bold,';
    } else if (this.fontWeight == "extrabold") {
      this.fontWeight = "800";
    } else if (this.fontWeight == "black") {
      this.fontWeight = "900";
    } else {
      this.fontWeight = "400";
      return "";
    }
    return `fontWeight: FontWeight.w${this.fontWeight},`;
  }
}

module.exports = { Text };

